Wikipedia/Code/WMFSearchResults.m

```diff
@@ -96,8 +96,13 @@ - (void)mergeRedirectMappingsFromModel:(WMFSearchResults *)searchResults {
 }
 
 - (void)mergeResultsFromModel:(WMFSearchResults *)searchResults {
+    NSMutableArray *displayTitlesOfExistingResults = [NSMutableArray new];
+    for (MWKSearchResult *obj in self.results) {
+        [displayTitlesOfExistingResults addObject:obj.displayTitle];
+    }
+    
     NSArray *newResults = [searchResults.results wmf_reject:^BOOL(MWKSearchResult *obj) {
-        return [self.results containsObject:obj];
+        return [displayTitlesOfExistingResults containsObject:obj.displayTitle];
```

Comment:
if `obj.displayTitle` is nil, `return YES;` and exclude the result from the new array. `displayTitle` _should_ never be nil, but technically it's `nullable`, so this is just to prevent a crash if that ever happens, since `containsObject:` and `addObject:` will crash when passed nil.